src/bin/tikv-ctl.rs

```diff
@@ -1416,3 +1496,44 @@ fn dump_snap_meta_file(path: &str) {
         );
     }
 }
+
+fn compact_whole_cluster(
+    pd: &str,
+    db_type: DBType,
+    cfs: Vec<&str>,
+    from: Option<Vec<u8>>,
+    to: Option<Vec<u8>>,
+    mgr: Arc<SecurityManager>,
+) {
+    let mut cfg = PdConfig::default();
+    cfg.endpoints.push(pd.to_owned());
+    if let Err(e) = cfg.validate() {
+        panic!("invalid pd configuration: {:?}", e);
+    }
+
+    let pd_client = RpcClient::new(&cfg, Arc::clone(&mgr))
+        .unwrap_or_else(|e| perror_and_exit("RpcClient::new", e));
+
+    let stores = pd_client
+        .get_all_stores()
+        .unwrap_or_else(|e| perror_and_exit("Get all cluster stores from PD failed", e));
+
+    let mut handles = Vec::new();
+    for s in stores {
+        let mgr = Arc::clone(&mgr);
+        let addr = s.address.clone();
+        let (from, to) = (from.clone(), to.clone());
+        let cfs: Vec<String> = cfs.iter().map(|cf| cf.to_string().clone()).collect();
+        let h = thread::spawn(move || {
+            let debug_executor = new_debug_executor(None, None, Some(&addr), None, mgr);
+            for cf in cfs {
+                debug_executor.compact(Some(&addr), db_type, cf.as_str(), from.clone(), to.clone());
```

Comment:
I mean, if you log here, you don't need to pass the address in.

Reply:
If log here, the `println` still can't be removed since there are `compact` and `compact_region` commands. It's better to use one `println` instead of two.